tests/__init__.py

```diff
@@ -359,3 +362,66 @@ def assert_url_equal(url1, url2):
     assert_equal(parts1.hostname, parts2.hostname)
     assert_equal(parts1.port, parts2.port)
     assert_equal(parse_qs(parts1.query), parse_qs(parts2.query))
+
+
+class RawResponse(BytesIO):
+    def stream(self, **kwargs):
+        contents = self.read()
+        while contents:
+            yield contents
+            contents = self.read()
+
+
+class BotocoreHTTPStubber(object):
+    def __init__(self):
+        self.reset()
+
+    def reset(self):
+        self._requests = []
+        self._responses = []
+
+    def add_response(self, response):
+        self._responses.append(response)
+
+    def create_response(self, url=None, status=200, headers=None, body=None):
+        if url is None:
+            url = 'https://example.com'
+
+        if headers is None:
+            headers = {}
+
+        if body is None:
+            body = b''
+
+        raw = RawResponse(body)
+        response = AWSResponse(url, status, headers, raw)
+        self._responses.append(response)
+
+    @property
+    def requests(self):
+        return self._requests
+
+    @property
+    def request_count(self):
```

Comment:
While this is not a public interface, I think it would be fine to just expose the `requests` property to keep the interface surface area small in case we ever want to make this abstraction public.